frontend/packages/client/src/contexts/NotificationModal.js

```diff
@@ -60,6 +60,8 @@ const NotificationModalProvider = ({ children }) => {
 
   const closeModal = useCallback(() => {
     setModal(false);
+    // this unmounts the component
+    setContent(null);
```

Comment:
by not setting content to null the component is still being referenced and not unmounted which causes next time to open the modal with the previous state of the last component used.